pkg/sentry/fs/proc/task.go

```diff
@@ -318,14 +323,14 @@ func (n *namespaceSymlink) Getlink(ctx context.Context, inode *fs.Inode) (*fs.Di
 
 	// Create a new regular file to fake the namespace file.
 	iops := fsutil.NewNoReadWriteFileInode(ctx, fs.RootOwner, fs.FilePermsFromMode(0777), linux.PROC_SUPER_MAGIC)
-	return fs.NewDirent(ctx, newProcInode(ctx, iops, inode.MountSource, fs.RegularFile, nil), n.Symlink.Target), nil
+	return fs.NewDirent(ctx, newProcNsInode(ctx, iops, inode.MountSource, fs.RegularFile, nil, n.inodeID), n.Symlink.Target), nil
 }
 
 func newNamespaceDir(t *kernel.Task, msrc *fs.MountSource) *fs.Inode {
 	contents := map[string]*fs.Inode{
-		"net":  newNamespaceSymlink(t, msrc, "net"),
-		"pid":  newNamespaceSymlink(t, msrc, "pid"),
-		"user": newNamespaceSymlink(t, msrc, "user"),
+		"net":  newNamespaceSymlink(t, msrc, "net", 0),
+		"pid":  newNamespaceSymlink(t, msrc, "pid", t.Pidns()),
```

Comment:
s/t.Pidns()/t.ThreadGroup().PIDNamespace().InodeNum() (then you don't need Task.Pidns() accessor).